src/api/models/assessment.py

```diff
@@ -403,8 +403,25 @@ class Assessment(BaseModel):
     )
     climatechange_monitored_text = models.TextField(blank=True)
 
+    # Disallow publishing if any fields are null. Does not check non-nullable fields with default values.
+    def clean(self):
+        if self.status == self.PUBLISHED:
```

Comment:
So this checks if its published? Does this run when you are publishing assessment or any time you save?

Reply:
Called by `full_clean` which is now called by `save` below, so yes in effect with every save on the model level.